web-docs/concepts/index.md

```diff
@@ -0,0 +1,190 @@
+
+Here are two flags, `flagOne` has `boolean` values and `flagTwo` has `string` values.
+
+### flags represented as JSON
+
+```json
+{
+  "flags": {
+    "flagOne": {
+      "state": "ENABLED",
+      "variants": {
+        "on": true,
+        "off": false
+      },
+      "defaultVariant": "on",
+      "targeting": {}
+    },
+    "flagTwo": {
+      "state": "ENABLED",
+      "variants": {
+        "key1": "val1",
+        "key2": "val2"
+      },
+      "defaultVariant": "key1",
+      "targeting": {}
+    }
+  }
+}
+```
+
+### flags represented as YAML
+
+```yaml
+flags:
+  flagOne:
+    state: ENABLED
+    variants:
+      'on': true
+      'off': false
+    defaultVariant: 'on'
+    targeting:
+  flagTwo:
+    state: ENABLED
+    variants:
+      key1: val1
+      key2: val2
+    defaultVariant: 'key1'
+    targeting:
+```
+
+## The structure of a flag
+
+Each flag has:
+
+- A flag key: `flagOne` and `flagTwo` above
+- A state: `ENABLED` or `DISABLED`
+- One or more possible `variants`. These are the possible values that a flag key can take.
+- An optional `targeting` rule (explained below)
+
+## Targeting rules
+
+Imagine you are introducing a new feature. You create a flag with two possible variants: `on` and `off`. You want to safely roll out the feature.
+Therefore the flags `defaultValue` is set to `off` for all users.
+
+In other words, the new feature is disabled by default.
+
+Now imagine you want to enable the feature, but only when both of these conditions are true:
+
+- Logged in users
+- The user email ends in `@example.com`
+
+Rather than codifying that in your application, flagd targeting rules can be used. The flag definition below models this behaviour.
+
+Your application is responsible for sending the `email` address via OpenFeature's context parameter (see below) and flagd will return the correct flag.
+
+```json
+{
+    "flags": {
+        "isFeatureEnabled": {
+            "state": "ENABLED",
+            "variants": {
+                "on": true,
+                "off": false
+                },
+            "defaultVariant": "off",
+            "targeting": {
+                "if": [{
+                    "in": [
+                        "@example.com",
+                        {
+                            "var": ["email"]
+                        }]
+                },
+                "on", null]
+            }
+         }
+    }
+}
+```
+
+### Pseudo-code of application passing context
+
+```js
+// The second parameter is the default in case flagd is unavailable
+featureAvailable = openFeature.getBooleanValue("isFeatureEnabled", false, {}) // false
+
+// isFeatureEnabled for a logged in user with an email example@gmail.com
+featureAvailable = openFeature.getBooleanValue("isFeatureEnabled", false, {"email": "example@gmail.com"}) // false
+
+// isFeatureEnabled for a logged in user with an email someone@example.com
```

Comment:
// Here, the flag is evaluated with targeting rule matching, hence the value of true